### PR TITLE
Testing, performance tests, and assorted bug fixes

### PR DESCRIPTION
**Pull request by jdfreder**
Sorry this is an all-in-one, but it was convenient to open the PR as-is.

This PR:
- fixes createStore initialState not setting the state if no actions were trigger
- fixes a bug in the fallback pwd, cwd call.  Missing parens
- listens to kernel status messages and updates the execution state accordingly (cc @betatim I this is kind of what you wanted)
- adds a performance API which is a noop if window.performance does not exist
- marks actions and agendas
- measures the execute_cell agenda
- adds basic performance test to travis
- adds an execute_cell agenda test
- adds convenience methods for testing against a live (kernel backed) state store

related to #422 , #128

### CODE REVIEW

File: src/notebook/actions/index.js [hunk 1]
```diff
@@ -28,9 +28,16 @@ export function newKernel(kernelSpecName, cwd) {
...
+        // Listen to the execution status of the kernel
+        channels.iopub
+          .filter(msg => msg.header.msg_type === 'status')
+          .map(msg => msg.content.execution_state)
+          .subscribe(state => subject.next(setExecutionState('idle')));
+
         agendas.acquireKernelInfo(channels)
               .subscribe(action => {
                 subject.next(action);
+                subject.next(setExecutionState('idle'));
```
Comment (rgbkrk) [inline src/notebook/actions/index.js:40]: :+1:

File: src/notebook/agendas/index.js [hunk 2]
```diff
@@ -141,11 +146,14 @@ export function executeCell(channels, id, code) {
...
          // Update the outputs with each change
          .subscribe(outputs => {
+           mark('executeCell:output');
+           measure('executeCell:roundtrip', 'executeCell:start','executeCell:output');
            subscriber.next(updateCellOutputs(id, outputs));
          })
     );
 
     shell.next(executeRequest);
+    mark('executeCell:end');
```
Comment (rgbkrk) [inline src/notebook/agendas/index.js:156]: End is a bit of a misnomer since this is more like "sent", the finish will be later.

**Comment by rgbkrk** [comment]
My hope was that we would send the current time stamp over with the actions, as would be useful for realtime, instead of using marks. I'm going to try this current implementation out soon though.

**Comment by jdfreder** [comment]
We can do that too, and remove the mark calls from everything but the agendas.  In the agendas we will keep the marks and measure call.

**Pull request merged**